internal/commands/user/delete/command.go

```diff
@@ -0,0 +1,164 @@
+package delete
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/commands/user/shared"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/10gen/realm-cli/internal/utils/flags"
+	"github.com/spf13/pflag"
+)
+
+// Command is the `user delete` command
+var Command = cli.CommandDefinition{
+	Use:         "delete",
+	Description: "Delete the user(s) from a Realm application",
+	Help:        "user delete",
+	Command:     &command{},
+}
+
+type command struct {
+	inputs      inputs
+	outputs     []output
+	realmClient realm.Client
+}
+
+type output struct {
+	user realm.User
+	err  error
+}
+
+func (cmd *command) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+
+	fs.StringSliceVarP(&cmd.inputs.Users, flagUsers, flagUsersShort, []string{}, flagUsersUsage)
+	fs.VarP(
+		flags.NewEnumSet(&cmd.inputs.ProviderTypes, shared.ValidProviderTypes),
+		shared.FlagProvider,
+		shared.FlagProviderShort,
+		shared.FlagProviderUsage,
+	)
+	fs.Lookup(shared.FlagProvider).NoOptDefVal = shared.ProviderTypeInteractive
+	fs.VarP(&cmd.inputs.State, shared.FlagStateType, shared.FlagStateTypeShort, shared.FlagStateTypeUsage)
+	fs.Lookup(shared.FlagProvider).NoOptDefVal = shared.ProviderTypeInteractive
```

Comment:
Ahh good catch. Should be trying to look at the FlagStateType. The lookup is to have reference to the flag that was created so that we can set the default value for when a value is not provided along with the flag. i.e. `--provider`

We talked about testing out the interactive without having an explicit interactive flag